src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/innerassignment/InputInnerAssignment.java

```diff
@@ -128,5 +128,47 @@ void method() throws IOException {
         while (line < 3 || (line = file.read()) != null) {}
         while (line < 3 & (line = file.read()) != null) {}
         while (line < 3 | (line = file.read()) != null) {}
+
+        do{}
+        while (!((line = file.read()) != null));
+        do{}
+        while ((line = file.read()) != null && line < 3);
+        do{}
+        while ((line = file.read()) != null && line < 3 && line > 5);
+        do{}
+        while ((line = file.read()) != null || line < 3);
+        do{}
+        while ((line = file.read()) != null || line < 3 || line > 5);
+        do{}
+        while ((line = file.read()) != null & line < 3);
+        do{}
+        while ((line = file.read()) != null & line < 3 & line > 5);
+        do{}
+        while ((line = file.read()) != null | line < 3);
+        do{}
+        while ((line = file.read()) != null | line < 3 | line > 5);
+        do{}
+        while (line < 3 && (line = file.read()) != null);
+        do{}
+        while (line < 3 || (line = file.read()) != null);
+        do{}
+        while (line < 3 & (line = file.read()) != null);
+        do{}
+        while (line < 3 | (line = file.read()) != null);
+
+        for (;!((line = file.read()) != null);) {}
+        for (;(line = file.read()) != null && line < 3;) {}
+        for (;(line = file.read()) != null && line < 3 && line > 5;) {}
+        for (;(line = file.read()) != null || line < 3;) {}
+        for (;(line = file.read()) != null || line < 3 || line > 5;) {}
+        for (;(line = file.read()) != null & line < 3;) {}
+        for (;(line = file.read()) != null & line < 3 & line > 5;) {}
+        for (;(line = file.read()) != null | line < 3;) {}
+        for (;(line = file.read()) != null | line < 3 | line > 5;) {}
+        for (;line < 3 && (line = file.read()) != null;) {}
+        for (;line < 3 || (line = file.read()) != null;) {}
+        for (;line < 3 & (line = file.read()) != null;) {}
+        for (;line < 3 | (line = file.read()) != null;) {}
+
```

Comment:
will it report violation for weird case like
```
String a = "";
for (;;(a = "das").concat()) {}
```
?

It can be added to test inputs

Reply:
> will it report violation for weird case like
> 
> ```
> String a = "";
> for (;;(a = "das").concat()) {}
> ```

@strkkk my patch is reporting violation for these kinds of cases, should these be omitted? ( I am not sure if this is a good way of assigning variables)

Reply:
This is definitely not the best idea to do like this.
I think it is ok to report violation in such case. Please add it to test inputs

Reply:
done